### ISSUE TITLE
Allow user to use `dataloader_idx` in log name in `LightningModule.log`

### ISSUE THREAD

**Issue by lokesh-vr-17773** [Lightning-AI/pytorch-lightning#20485]
### Description & Motivation

Allow the user to include dataloader information in the log name when `add_datalaoder_idx` is set to False.

### Pitch

Since `LightningModule.log` includes the `add_dataloader_idx` flag for logging, it is beneficial to allow users to include dataloader information in the log name when `add_dataloader_idx` is set to `False`. This aligns with the intended purpose of the `add_dataloader_idx` flag.

Proposed change:
```python
if add_dataloader_idx and "/dataloader_idx_" in name:
  raise MisconfigurationException(
      f"You called `self.log` with the key `{name}`"
      " but it should not contain information about `dataloader_idx`"
  )
```

**Comment by lantiga** [comment]
Makes sense, would you like to send in a quick PR? Also it would be great if you could summarize why this is important for you as a user by providing a real-world case.
